integration/experiment/gpu_frequency_sweep/gpu_frequency_sweep.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2022, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+'''
+Helper functions for running gpu frequency sweep experiments.
+'''
+
+import sys
+import argparse
+import os
+
+import geopmpy.agent
+
+from experiment import launch_util
+from experiment import common_args
+from experiment import machine
+from experiment.frequency_sweep import frequency_sweep
+from experiment.uncore_frequency_sweep import uncore_frequency_sweep
+
+def setup_run_args(parser):
+    uncore_frequency_sweep.setup_run_args(parser)
+    parser.add_argument('--max-gpu-frequency', dest='max_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='top gpu frequency setting for the sweep')
+    parser.add_argument('--min-gpu-frequency', dest='min_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='bottom gpu frequency setting for the sweep')
+    parser.add_argument('--step-gpu-frequency', dest='step_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='increment in hertz between gpu frequency settings for the sweep')
+
+def setup_gpu_frequency_bounds(mach, min_gpu_freq, max_gpu_freq,
+                                  step_gpu_freq):
+    sys_min = mach.gpu_frequency_min()
+    sys_max = mach.gpu_frequency_max()
+    #TODO: Replace with gpu step when signal is merged in
+    sys_step = mach.frequency_step()
+    if min_gpu_freq is None:
+        min_gpu_freq = sys_min
+    if max_gpu_freq is None:
+        max_gpu_freq = sys_max
+    if step_gpu_freq is None:
+        step_gpu_freq = sys_step
+    if min_gpu_freq < sys_min or max_gpu_freq > sys_max:
+        raise RuntimeError('GPU frequency bounds are out of range for this system')
+    if min_gpu_freq > max_gpu_freq:
+        raise RuntimeError('GPU frequency min is greater than max')
+
+    gpu_freqs = reversed(experiment_steps(min_gpu_freq, max_gpu_freq, step_gpu_freq))
+    return gpu_freqs
+
+def experiment_steps(minv, maxv, stepv):
+    rval = [minv]
+    while rval[-1] + stepv <= maxv:
+        rval.append(rval[-1] + stepv)
+    return rval
+
+def report_signals():
+    return ["CPU_CYCLES_THREAD@package", "CPU_CYCLES_REFERENCE@package",
+            "TIME@package", "CPU_ENERGY@package", "GPU_CORE_FREQUENCY_STATUS@board",
+            "GPU_CORE_FREQUENCY_MAX_CONTROL@board"]
```

Comment:
This doesn't work as expected--reading GPU_CORE_FREQUENCY_MAX_CONTROL is static, even if it has been written to.

Reply:
Does that also happen for `GPU_CORE_FREQUENCY_MAX_CONTROL@gpu_chip`?

Reply:
The aggregate issue is gone in that case, but it is still giving static info, so it's not a useful signal.

Reply:
Try with latest commit using @board.